class-udm-updater.php

```diff
@@ -103,6 +103,36 @@ public function __construct($mothership, $muid, $relative_plugin_file, $options
 		if (version_compare($wp_version, '5.5', '<')) {
 			add_filter('auto_update_plugin', array($this, 'auto_update_plugin'), 20, 2);
 		}
+
+		add_action('wp_loaded', array($this, 'maybe_force_checking_for_updates'));
+	}
+
+	/**
+	 * Whether to manually force checking for updates by inspecting a specified slug submitted as a query argument (force_udm_check) and build restriction where plugin can only be force checked 2 times a day to prevent abuse
+	 *
+	 * @return Void
+	 */
+	public function maybe_force_checking_for_updates() {
+		if (!isset($_REQUEST['force_udm_check']) || '' === $_REQUEST['force_udm_check']) return;
+		$slug = sanitize_text_field($_REQUEST['force_udm_check']);
+		if ($this->slug !== $slug) return;
+		$restriction = get_transient('udm_manual_check_for_updates_restriction');
```

Comment:
@andtown This is a mis-use of the transients API. The transients API is for caching of data that can always be re-created. It should always be assumed that `get_transient()` can fail - the transient backend is allowed to delete anything and everything, any time. It just so happens that the default backend (MySQL) doesn't do this.... and because of that, lots of developers assume that it can be relied upon - but that's not what the API itself promises. If you want to be sure that things saved to it will ever be available, use an option. (You can store the current day in the option and check its value).

Reply:
@DavidAnderson684 Thanks for reminding me about this one. I remember there was a discussion regarding transients on the #development channel on Slack. The Trello card says about using a transient, but I just realised that we can't use it for this case, and since options API is the solution for this, so I've used options API instead of transients.